modules/sage/1.0/sage.smk

```diff
@@ -0,0 +1,255 @@
+          &&
+        wget -O {output.panel_bed} {params.url}/ActionableCodingPanel.somatic.{params.build}.bed.gz
+          &&
+        wget -O {output.high_conf_bed} {params.url}/HighConfidence.{params.build}.bed.gz
+        """)
+
+# Non-standard chromosomes in rare cases cause SAGE error. This function will read the main chromosomes
+# file for each genome build using file produced by reference_files workflow, and supply it as
+# a comma-deliminated list of chromosomes for SAGE run.
+def get_chromosomes(wildcards):
+    path = reference_files("genomes/"+str(wildcards.genome_build)+"/genome_fasta/main_chromosomes.txt")
+    input = str(path)
+    with open(input, 'r') as chromosome_file:
+        lines = chromosome_file.readlines()
+        chromosomes=[]
+        for x in lines:
+            chromosomes.append(x.rstrip("\n"))
+        chromosomes= ",".join(chromosomes)
+    return chromosomes
+
+# Variant calling rule
+rule _run_sage:
+    input:
+        tumour_bam = CFG["dirs"]["inputs"] + "bam/{seq_type}--{genome_build}/{tumour_id}.bam",
+        normal_bam = CFG["dirs"]["inputs"] + "bam/{seq_type}--{genome_build}/{normal_id}.bam",
+        fasta = str(rules._input_references.output.genome_fa),
+        hotspots = str(rules._download_sage_references.output.hotspots),
+        panel_bed = str(rules._download_sage_references.output.panel_bed),
+        high_conf_bed = str(rules._download_sage_references.output.high_conf_bed)
+    output:
+        vcf = temp(CFG["dirs"]["sage"] + "{seq_type}--{genome_build}/{tumour_id}--{normal_id}--{pair_status}/{tumour_id}--{normal_id}--{pair_status}.vcf"),
+        vcf_gz = CFG["dirs"]["sage"] + "{seq_type}--{genome_build}/{tumour_id}--{normal_id}--{pair_status}/{tumour_id}--{normal_id}--{pair_status}.vcf.gz"
+    log:
+        stdout = CFG["logs"]["sage"] + "{seq_type}--{genome_build}/{tumour_id}--{normal_id}--{pair_status}/run_sage.stdout.log",
+        stderr = CFG["logs"]["sage"] + "{seq_type}--{genome_build}/{tumour_id}--{normal_id}--{pair_status}/run_sage.stderr.log"
+    params:
+        opts = CFG["options"]["sage_run"],
+        chromosomes = get_chromosomes,
+        assembly = lambda w: "hg38" if "38" in str({w.genome_build}) else "hg19",
+        sage= "$(dirname $(readlink -e $(which SAGE)))/sage.jar",
+        jvmheap = lambda wildcards, resources: int(resources.mem_mb * 0.8)
+    conda:
+        CFG["conda_envs"]["sage"]
+    threads:
+        CFG["threads"]["sage_run"]
+    resources:
+        **CFG["resources"]["sage_run"]
+    shell:
+        op.as_one_line("""
+        echo "running {rule} for {wildcards.tumour_id}--{wildcards.normal_id} on $(hostname)" > {log.stdout}
+        &&
+        java -Xms1G -Xmx{params.jvmheap}m
+        -cp {params.sage} com.hartwig.hmftools.sage.SageApplication
+        -threads {threads}
+        {params.opts}
+        -chr {params.chromosomes}
+        -reference {wildcards.normal_id}
```

Comment:
I was puzzled by this at first. Reference is the name for the normal!

Reply:
Yes it is confusing to me as well! Why not name it `normal`, if it is fact not reference, but the normal sample.